### DISCUSSION TITLE
Box Conductor

### DISCUSSION THREAD

**Discussion by farhoud** [functionland/box#141] [Ideas]
We have a lot of component:
- Box
- IPFS-cluster
- IPFS
- Disk Management

That need process,config and update management. me and @keyvan-m-sadeghi idea was to use docker and docker-compose as process manager.
In short the Conductor will be an application which expose config API and manage docker and docker-compose.

I think we should start designing it, and and need your collaboration.
@gitaaron 
@masih 
@ehsan6sha 
@keyvan-m-sadeghi 
@mehdibalouchi 
@matiqorb 
@ghorbani-m

**Comment by gitaaron** [comment]
@farhoud  mentioned he'd like to formulate a requirements / specs doc.

These may be used to assist with certain requirements here -

- https://nixos.org/
- https://earthly.dev/
- https://www.ansible.com/

@masih  - FYI - eyeballs on this are appreciated

**Comment by gitaaron** [comment]
In his [writeup](https://www.snowfreakz.nl/functionland-diy-box-with-raspberry-pi/), @joomapp mentions a few other tools related to this -

- [Portainer](https://www.portainer.io/)
- [Watchdog](https://linux.die.net/man/8/watchdog)
